lib/Conversion/TorchToLinalg/TensorConstructors.cpp

```diff
@@ -181,12 +181,17 @@ class ConvertAtenEmptyMemoryFormatOp
   LogicalResult
   matchAndRewrite(AtenEmptyMemoryFormatOp op, OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
+
     if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
       return failure();
 
     // TODO: Add support for layout, pin_memory and memory_format features.
-    // Only `none` layout is supported.
-    if (!op.layout().getType().template isa<Torch::NoneType>())
+    // Only 'strided' and `none` layout is supported. The memory layout should
+    // be `none` or `contiguous` to make sure that default strides are used.
+    int64_t layout;
+    if (!op.layout().getType().template isa<Torch::NoneType>() &&
+        (!matchPattern(op.layout(), m_TorchConstantInt(&layout)) ||
+         layout != torch_upstream::Layout::Strided))
```

Comment:
the last check `layout !=torch_upstream::Layout::Strided` should only be executed when `matchPattern(op.layout(), m_TorchConstantInt(&layout)` succeed. You can refer to the code here:
https://github.com/llvm/torch-mlir/blob/5b9bdfaf3fb56c5aed12d689e21c9884b1c5a789/lib/Conversion/TorchToTMTensor/TorchToTMTensor.cpp#L205-L213. There are multiple other places in this file with similar issue. Can you help to fix those as well? Maybe add helps like `checkIfNoneOrConstantbool`, `checkIfNoneOrConstantInt` etc

Reply:
Okay sure.

Reply:
As suggested by sean, I am removing the layout check.